### PR TITLE
Haar 2038 filter functionality

### PR DESCRIPTION
**Pull request by thomasridd**
### What
- **HAAR-2038** Functionality for filtering the main homepage + tests
- **HAAR-1910** delete client instance flow - journey to add duplicate client instance to base clients - includes route, controller, template and presenter utils
- **HAAR-1909** add client instance flow - journey to add duplicate client instance to base clients - includes route, controller, template and presenter utils
- remove legacy Helm files
- update docker image to include new manage users api

### CODE REVIEW

File: docker-compose.yml
Comment (thomasridd) [inline docker-compose.yml:1]: Bugfix for merging in hmpps-typescript-template changes

File: helm_deploy/hmpps-template-typescript/.helmignore
Comment (thomasridd) [inline helm_deploy/hmpps-template-typescript/.helmignore:1]: Bugfix for merging in hmpps-typescript-template changes

File: helm_deploy/hmpps-template-typescript/values.yaml
Comment (thomasridd) [inline helm_deploy/hmpps-template-typescript/values.yaml:1]: Bugfix for merging in hmpps-typescript-template changes

File: server/controllers/baseClientController.test.ts
Comment (thomasridd) [inline server/controllers/baseClientController.test.ts:1]: HAAR-1909 - Create client instance

File: server/controllers/baseClientController.test.ts [hunk 1]
```diff
@@ -266,4 +266,132 @@ describe('BaseClientController', () => {
...
+      // WHEN it is posted
+      await baseClientController.createClientInstance()(request, response, next)
+
+      // THEN the new base client success page is rendered
+      expect(response.render).toHaveBeenCalledWith(
+        'pages/new-base-client-success.njk',
+        expect.objectContaining({ secrets }),
+      )
+    })
+  })
```
Comment (thomasridd) [inline server/controllers/baseClientController.test.ts:289]: [HAAR-1909](https://dsdmoj.atlassian.net/browse/HAAR-1909) - Create client instance

File: server/controllers/baseClientController.ts [hunk 2]
```diff
@@ -23,6 +24,21 @@ export default class BaseClientController {
...
+      const baseClients = await this.baseClientService.listBaseClients(userToken)
+
+      const presenter = listBaseClientsPresenter(baseClients, filter)
+
+      res.render('pages/base-clients.njk', {
+        presenter,
+      })
+    }
+  }
+
```
Comment (thomasridd) [inline server/controllers/baseClientController.ts:41]: HAAR-2308 filter functionality

File: server/controllers/baseClientController.ts [hunk 3]
```diff
@@ -159,4 +175,78 @@ export default class BaseClientController {
...
+      // Display delete confirmation page
+      res.render('pages/delete-client-instance.njk', {
+        baseClient,
+        clientId,
+        isLastClient: clients.length === 1,
+        error,
+      })
+    }
+  }
```
Comment (thomasridd) [inline server/controllers/baseClientController.ts:217]: [HAAR-1909](https://dsdmoj.atlassian.net/browse/HAAR-1909) - Create client instance

File: server/controllers/baseClientController.ts [hunk 4]
```diff
@@ -159,4 +175,78 @@ export default class BaseClientController {
...
+      await this.baseClientService.deleteClientInstance(userToken, client)
+
+      // return to view base client screen (or home screen if last client deleted)
+      if (clients.length === 1) {
+        res.redirect(`/`)
+      } else {
+        res.redirect(`/base-clients/${baseClientId}`)
+      }
+    }
+  }
```
Comment (thomasridd) [inline server/controllers/baseClientController.ts:251]: [HAAR-1910](https://dsdmoj.atlassian.net/browse/HAAR-1910) - Delete client instance

File: server/interfaces/baseClientApi/baseClient.ts
Comment (thomasridd) [inline server/interfaces/baseClientApi/baseClient.ts:1]: HAAR 2308 - Filter functionality

File: server/mappers/forms/mapFilterForm.ts
Comment (thomasridd) [inline server/mappers/forms/mapFilterForm.ts:1]: HAAR-2308 Filter functionality

File: server/views/pages/base-clients.njk
Comment (thomasridd) [inline server/views/pages/base-clients.njk:1]: HAAR-2308 Filter functionality - make filter values persist after filters applied

File: server/views/pages/delete-client-instance.njk
Comment (thomasridd) [inline server/views/pages/delete-client-instance.njk:1]: HAAR-1910 Delete client confirmation page

File: server/views/presenters/listBaseClientsPresenter.test.ts
Comment (thomasridd) [inline server/views/presenters/listBaseClientsPresenter.test.ts:1]: HAAR-2308 Filter functionality

File: server/views/presenters/listBaseClientsPresenter.ts
Comment (thomasridd) [inline server/views/presenters/listBaseClientsPresenter.ts:1]: HAAR-2308 Filter functionality - main functionality

**Review by SimonMitchellMOJ — APPROVED** [review]
lgtm

**Pull request merged**